lib/puppet.rb

```diff
@@ -249,11 +249,13 @@ def self.base_context(settings)
       loaders = Puppet::Environments::Directories.from_path(environmentpath, basemodulepath)
       # in case the configured environment (used for the default sometimes)
       # doesn't exist
+      modulepath = settings[:modulepath]
+      modulepath = modulepath.nil? ? basemodulepath : Puppet::Node::Environment.split_path(modulepath)
```

Comment:
I agree with your interpretation of the docs - was asking question more as devil's advocate / is there anything that might break based on prior behavior.